### PR TITLE
recursive calculation of underlying protocol tokens

### PR DESCRIPTION
**Pull request by bergarces**
- Added method to AdaptersController to create a list of protocol tokens per chain
- Added decorator that recursively computes underlying positions
- Fixed wrong metadata symbol for lido protocol tokens.

### CODE REVIEW

File: src/adapters/lido/products/st-eth/lidoStEthAdapter.ts
Comment (bergarces) [inline src/adapters/lido/products/st-eth/lidoStEthAdapter.ts:50]: Using `getTokenMetadata` to avoid mismatches with our metadata list.

File: src/core/adaptersController.ts
Comment (bergarces) [inline src/core/adaptersController.ts:85]: This code cannot be added until the newreward type is merged.

File: src/types/IProtocolAdapter.ts
Comment (bergarces) [inline src/types/IProtocolAdapter.ts:39]: As discussed, I think it would be beneficial to stop adding things here at public level and add instead a single toolbox class that exposes what we need:
- Methods from adapter controller
- ChainId
- Provider
- Anything new we want to add

If we do that, then we won't have to change every adapter that implements this interface directly every time we add a utility to that class (unless that adapter needs to use it).

**Review by jpsains — APPROVED** [review]
LGTM I uncommented the if(reward) continue

**Pull request merged**
